fastapi_jsonapi/data_layers/base.py

```diff
@@ -80,9 +80,8 @@ def _apply_client_generated_id(
         """
         if data_create.id is None:
             return model_kwargs
-
-        extra = data_create.__fields__["id"].field_info.extra
-        if extra.get("client_can_set_id"):
+        extra = data_create.model_fields["id"].default
```

Comment:
extra != default
default is what's set as default if data not passed
extra is extra kwargs

```python
class Foo(BaseModel):
    # "John" is default for the name field
    # {"foo": "bar"} is field's extra
    name: str = Field("John", foo="bar")
```